FrontEnd/src/App.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import './index.css';
+import Recaptcha from 'react-recaptcha';
+
+export default class App extends Component {
+  render() {
+    return (
+      <Register />
+    );
+  }
+}
+
+// eslint-disable-next-line
+const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+
+const validateForm = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach(
+      (val) => val.length > 0 && (valid = false)
+    );    
+    return valid;
+}
+
+class Register extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.recaptchaLoaded = this.recaptchaLoaded.bind(this);
+    this.verifyCallback = this.verifyCallback.bind(this);
+    
+    this.state = {
+      firstName: null,
+      email: null,
+      message: null,
+      isVerified: false,
+      errors: {
+        firstName: '',
+        lastName: '',
+        email: '',
+        message: ''
+      }
+    };
+  }
+
+  handleChange = (event) => {
+    event.preventDefault();
+    const { name, value } = event.target;
+    let errors = this.state.errors;
+
+    switch (name) {
+      case 'firstName': 
+        errors.firstName = value.length == 0 ? 'Please enter your first name' : '';
+        break;
+
+        case 'lastName': 
+        errors.lastName = value.length == 0  ? 'Please enter your last name'  : '';
+        break;
+      case 'email': 
+        errors.email = validEmailRegex.test(value) ? '' : 'Please enter a valid email address';
+        break;
```

Comment:
This seems like it will make an error message pop up whenever the user types anything into the email box, until they type a correct email address. How would you make it so that the user only sees an error after clicking away?

Reply:
I'd remove the OnChange event. Then, change the error requirements so that it first needs to detect an OnClick and have it validate on an OnClickOut (or something similar). 

The only thing is I'm not sure if the OnClickOut would trigger before the Submit button's OnClick (assuming the rest of the form was done correctly and the email's data is invalid)

Reply:
You can probably just assign this validation to an onBlur (what you called onClickOut), no need to check for a click first. But yea, it should be when the user leaves the input. 👍

Reply:
Well then, onBlur just makes things really easy!  Good to know for my next project.